diag_manager/fms_diag_field_object.F90

```diff
@@ -964,5 +965,78 @@ PURE FUNCTION diag_field_id_from_name(this, module_name, field_name) &
   endif
 end function diag_field_id_from_name
+
+!> Dumps any data from a given fmsDiagField_type object
+subroutine dump_field_obj (this, unit_num)
+  class(fmsDiagField_type), intent(in) :: this
+  integer, intent(in) :: unit_num !< passed in from dump_diag_obj if log file is being written to
+  integer :: i
+
+  if( mpp_pe() .eq. mpp_root_pe()) then
+    if( allocated(this%file_ids)) write(unit_num, *) 'file_ids:' ,this%file_ids
+    if( allocated(this%diag_id)) write(unit_num, *) 'diag_id:' ,this%diag_id
+    if( allocated(this%static)) write(unit_num, *) 'static:' ,this%static
+    if( allocated(this%registered)) write(unit_num, *) 'registered:' ,this%registered
+    if( allocated(this%mask_variant)) write(unit_num, *) 'mask_variant:' ,this%mask_variant
+    if( allocated(this%do_not_log)) write(unit_num, *) 'do_not_log:' ,this%do_not_log
+    if( allocated(this%local)) write(unit_num, *) 'local:' ,this%local
+    if( allocated(this%vartype)) write(unit_num, *) 'vartype:' ,this%vartype
+    if( allocated(this%varname)) write(unit_num, *) 'varname:' ,this%varname
+    if( allocated(this%longname)) write(unit_num, *) 'longname:' ,this%longname
+    if( allocated(this%standname)) write(unit_num, *) 'standname:' ,this%standname
+    if( allocated(this%units)) write(unit_num, *) 'units:' ,this%units
+    if( allocated(this%modname)) write(unit_num, *) 'modname:' ,this%modname
+    if( allocated(this%realm)) write(unit_num, *) 'realm:' ,this%realm
+    if( allocated(this%interp_method)) write(unit_num, *) 'interp_method:' ,this%interp_method
+    if( allocated(this%tile_count)) write(unit_num, *) 'tile_count:' ,this%tile_count
+    if( allocated(this%axis_ids)) write(unit_num, *) 'axis_ids:' ,this%axis_ids
+    write(unit_num, *) 'type_of_domain:' ,this%type_of_domain
+    if( allocated(this%area)) write(unit_num, *) 'area:' ,this%area
+    if( allocated(this%missing_value)) then
+      select type(missing_val => this%missing_value)
+        type is (real(r4_kind))
+          write(unit_num, *) 'missing_value:', missing_val
+        type is (real(r8_kind))
+          write(unit_num, *) 'missing_value:' ,missing_val
+       type is(integer(i4_kind))
+          write(unit_num, *) 'missing_value:' ,missing_val
+       type is(integer(i8_kind))
+          write(unit_num, *) 'missing_value:' ,missing_val
+      end select
+    endif
+    if( allocated( this%data_RANGE)) then
+      select type(drange => this%data_RANGE)
+        type is (real(r4_kind))
+          write(unit_num, *) 'data_RANGE:' ,drange
+        type is (real(r8_kind))
+          write(unit_num, *) 'data_RANGE:' ,drange
+       type is(integer(i4_kind))
+          write(unit_num, *) 'data_RANGE:' ,drange
+       type is(integer(i8_kind))
+          write(unit_num, *) 'data_RANGE:' ,drange
+      end select
+    endif
+    write(unit_num, *) 'num_attributes:' ,this%num_attributes
+    if( allocated(this%attributes)) then
+      do i=1, SIZE(this%attributes)
```

Comment:
I think this should be `this%num_attributes`
this%attributes is allocated to be
https://github.com/NOAA-GFDL/FMS/blob/bd4c4c6ac6d7d0bcedcd92eca09a6033c15c9ee2/diag_manager/fms_diag_field_object.F90#L319

but the field can have less attributes than that

Reply:
I'm pissed cause i had that originally lol, thought getting the allocated size would be better